src/main/java/seedu/address/logic/commands/AddRecordCommand.java

```diff
@@ -0,0 +1,92 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CONDITION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.record.Record;
+
+/**
+ * Adds a record to a patient in the MedBook
+ */
+public class AddRecordCommand extends Command {
+
+    public static final String COMMAND_WORD = "addrecord";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a patient record to the address book. "
+            + "Parameters: "
+            + PREFIX_DATE + "DATE "
+            + PREFIX_CONDITION + "CONDITION " + "\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_DATE + "18092023 1800"
+            + PREFIX_CONDITION + "Fever";
+
+    public static final String MESSAGE_SUCCESS = "New record added";
+
+    private final Record record;
+
+    private final Index index;
+
+    /**
+     * Creates a record under a patient of specified index
+     */
+    public AddRecordCommand(Index index, Record record) {
+        requireAllNonNull(index, record);
+        this.index = index;
+        this.record = record;
+    }
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToAddRecord = lastShownList.get(index.getZeroBased());
+        List<Record> newRecords = new ArrayList<>(personToAddRecord.getRecords());
+        newRecords.add(record);
+        Person personWithAddedRecord = new Person(personToAddRecord.getName(), personToAddRecord.getEmail(),
```

Comment:
Should add a check to see if the existing Record already exists to prevent duplicate Records

Reply:
Created UniqueRecordList to prevent duplicate records.